api/client/webclient/webconfig.go

```diff
@@ -67,6 +67,8 @@ type WebConfig struct {
 	AutomaticUpgrades bool `json:"automaticUpgrades"`
 	// AssistEnabled is true when Teleport Assist is enabled.
 	AssistEnabled bool `json:"assistEnabled"`
+	// FeatureHidingEnabled is true when features should be undiscoverable to users without the necessary permissions.
```

Comment:
I'd explain that usually we show certain UI elements even if the user doesn't have access to them, but with this flag being set to true, the elements are hidden instead. "Features should be undiscoverable" doesn't explain how we make them undiscoverable.